Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -84,6 +84,32 @@ open class NavigationMapView: UIView {
         }
     }
     
+    /**
+     Controls whether the main route style layer and its casing disappears as the user location puck travels over it. Defaults to `false`.
+     
+     Used in standalone `NavigationMapView` during navigation. If using `NavigationViewController` and `CarPlayNavigationViewController`
+     for active navigation, update `NavigationViewController.routeLineTracksTraversal` and `CarPlayNavigationViewController.routeLineTracksTraversal` instead.
+     
+     If `true`, the part of the route that has been traversed will be
+     rendered with full transparency, to give the illusion of a
+     disappearing route. To customize the color that appears on the
+     traversed section of a route, override the `traversedRouteColor` property
+     for the `NavigationMapView.appearance()`. If `false`, the whole route will be shown without traversed
+     part disappearing effect.
+     
+     To update the route line during active navigation when `RouteProgree` changes, add observer for `Notification.Name.routeControllerProgressDidChange` and
+     call `NavigationMapView.updateRouteLine(routeProgress:coordinate:shouldRedraw:)` with `shouldRedraw` as `false`.
+     
+     To update the route line during active navigation when route refresh or rerouting happens, add observers for `Notification.Name.routeControllerDidRefreshRoute` and
+     `Notification.Name.routeControllerDidReroute`. And call `NavigationMapView.updateRouteLine(routeProgress:coordinate:shouldRedraw:)`
+     with `shouldRedraw` as `true`.
```

Comment:
I still think that we may need to leave the notification handling to users. Because when using standalone `NavigationMapView ` for active navigation, they already subscribed the notifications of routeProgress change, route refresh or rerouting events, and make updates to the waypoints, route line, user location, etc. If we make the `NavigationMapView` default to handle the route line update only in active navigation, it may cause issue with their previous implementation. It may be more reasonable and stable to users by providing the `NavigationMapView.updateRouteLine(routeProgress:coordinate:shouldRedraw:)` method to use to update route line when they need it.